doc/dev/kdb-operations.md

```diff
@@ -1,9 +1,11 @@
 # Proposal: KDB Operations
 
 There are four main _operations_ in `libelektra-kdb`: `open`, `get`, `set` and `close`.
-For each of these there is a `kdb*` function the user calls to trigger the operation and plugins implement a `elektra<Plugin>*` function for each of the operations they support (at least `get`).
+For each of these there is a `kdb*` function the user calls to trigger the operation and plugins export a function for each of the operations they support (at least `get`).
 
-Additionally, plugins may implement `elektra<Plugin>Commit` and `elektra<Plugin>Error`.
+<!-- TODO: should commit and error really be separate? With the new elektraPluginGetPhase they could just be part of set... -->
```

Comment:
@markus2330 What do you think? IMO `commit`/`error` should be merged back into `set`.

The current solution just creates unnecessary extra code in both `libelektra-kdb`

https://github.com/ElektraInitiative/libelektra/blob/fac849659be65b3021ded570dcf33e2edb95c407/src/libs/elektra/kdb.c#L2028-L2035

and backend plugins

https://github.com/ElektraInitiative/libelektra/blob/fac849659be65b3021ded570dcf33e2edb95c407/src/plugins/backend/backend.c#L562-L589

(there is more and it is and is all basically copy-paste from the set code)

Reply:
This is up to @flo91 to decide. I thought it is a good idea to have commit+error, in particular for the database backend.

Reply:
I don't think that code duplication in backend plugins should be the decisive factor here. IMHO more important is the clarity in resolvers. But I am open to both ways, having plugins with only open/get/set/close is more elegant.

Reply:
Another (more dramatic) option would be to add the current phase as a 4th parameter to the `get` and `set` functions of plugins:

```c
int elektraFooGet (Plugin * handle, ElektraGetPhase phase, KeySet * ks, Key * parentKey);
int elektraFooSet (Plugin * handle, ElektraSetPhase phase, KeySet * ks, Key * parentKey);
```

The it's also quite clear that `phase` should probably be handled in some way.

Reply:
No, the phase can be communicated via the plugin handle better. Not every plugin is interested in the phase. In any case: there is doc/decisions/commit_function.md, which needs an update.

Reply:
> there is doc/decisions/commit_function.md, which needs an update.

Sure, but do we want a separate commit/error function or not. Because if not, I would just remove the file. I can create a new `kdb_phases.md` decisison, but I'm not sure we need it. It's already implemented and it is documented in `doc/dev/backend-plugins.md`